src/main/java/org/terasology/blockNetwork/NetworkNode.java

```diff
@@ -54,7 +80,12 @@ public NetworkNode(Vector3i location, byte inputSides, byte outputSides) {
         this.inputSides = inputSides;
         this.outputSides = outputSides;
     }
-
+    
+    /**
+     * Used for finding out whether or not two NetworkNodes are the same
+     * @param o The node being compared
+     * @return true only if the node being compared to has the same inputSides, outputSides, and location
+     */
```

Comment:
Usually, following the DRY (Don't Repeat Yourself) principle, we tend to not add javadocs for equals, toString and the like (since the javadocs are inherited from the parent function that you are overriding). However, since we don't want to lose the information on equality of two nodes, add this to the class-level javadoc:
`Note that two Nodes are are considered equal if ...`